lib/hawkular/base_client.rb

```diff
@@ -174,14 +174,16 @@ def tenant_header
     end
 
     def handle_fault(f)
+      http_code = (f.respond_to?(:http_code) ? f.http_code : 0)
+      fail HawkularException.new('Invalid credentials', http_code) if f.instance_of? RestClient::Unauthorized
```

Comment:
@abonas this is an issue in WildFly - it will return HTML no matter what accepts header you send it. Which is why this PR exists at all. 
@rubenvp8510 parsing the body as JSON will not work. A simple regex could in this case 

401 Unauthorized:
`<html><head><title>Error</title></head><body>Unauthorized</body></html>`

500 with accept type of json:
```
{
  "errorMsg" : "RESTEASY003210: Could not find resource for full path: http://localhost:8080/hawkular/inventory/foo"
}
```

What happens is either a message is unauthorized and we get a 401+html from the appserver itself. Or it is authorized and then passed to the rest-layer (which is part of the application) and this then correctly handles the response types (and returns json).